Tests/iaas/mandatory-services/mandatory-iaas-services.py

```diff
@@ -0,0 +1,96 @@
+"""Mandatory APIs checker
+
+This script retrieves the endpoint catalog from Keystone using the OpenStack
+SDK and checks whether all mandatory APi endpoints, are present.
+The script relies on an OpenStack SDK compatible clouds.yaml file for
+authentication with Keystone.
+As the s3 endpoint might differ, a missing one will only result in a warning.
+"""
+
+import argparse
+import logging
+import os
+
+import openstack
+
+
+logger = logging.getLogger(__name__)
+mandatory_services = ["compute", "identity", "image", "block-storage",
+                      "network", "load-balancer", "placement"]
+object_store_service = ["s3", "object-store"]
```

Comment:
I have never seen a single cloud having S3 service/endpoint in the catalog. Typically `object-store` is also supporting S3 (either it is swift with S3 enabled or it is rgw catalog pointing to swift endpoint).

Reply:
There is no tooling existing in the world (known to me) that expects S3 and consumes OpenStack service catalog searching for "s3" service type